### PR TITLE
🐛 Fix ExclusiveSplitGateway onError state change

### LINKED ISSUE #158: 🐛 ExclusiveSplitGateways do not persists Error-states
#### Describe your issue

The ExclusiveSplitGateway performs various preflight checks to ensure it can take the correct branch (i.e. checking if one - and only one - outgoing SequenceFlow has a truthy condition attached to it).
When these checks fail, the Gateway is supposed to throw an Error, telling the user what preflight check has failed.

Currently, these errors are not persisted, resulting in the Gateway to remain in an active state after encountering an error.
This leads to faulty behavior, as the user gets stuck in the execution of the process that the gateway in question belongs to.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

## Example

<details>
<summary>
:warning: Before creating the issue, click here to expand and follow the guide.
</summary>
<br>
  
1. Be sure to fill out **every** topic presented in this template. 
If a section is not valid or useful for your issue, feel free to remove it.

2. Use the attached Checklist to make sure your issue contains all relevant information.

3. Attach a label to your issue that reflects what your issue is all about.
**When your issue concerns only a specific platform, be sure to attach the label for that platform!**

4. Please use a fitting emoji at the beginning of your issue title.

Example 1 - Report a bug:
```md
:bug: Something is not working properly
```

Example 2 - Report a performance issue:
```md
:zap: Something is slower than it should be
```

Example 3 - Report a package vulnerability or security issue:
```md
:lock: Package xy produces vulnerabilites
```

```md
:lock: User XY has access to something he shouldn't have
```

Example 4 - Report missing test coverage: 
```md
:white_check_mark: Add tests for...
```

You can of course phrase your titles however you like.
Just be sure they are short, to the point and convey the general idea of the issue you are experiencing.
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your Issue with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

Fix an issue, where ExclusiveSplitGateways that encountered an error did not have their state changed accordingly.

**Issues:**

Closes #158 

PR: #159

## How can others test the changes?

Run Processes with erroneously configured ExclusiveSplitGateways and see that these will no be persisted in an "error" state when they are run.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: test/1_process_engine/interprocess_communication_test.js [hunk 1]
```diff
@@ -45,7 +49,7 @@ describe('Inter-process communication - ', () => {
     // As a result we must subscribe to the event that gets send when the test is done.
     testFixtureProvider.executeProcess(processModelReceiveEvents, 'MessageStartEvent_1', correlationId);
 
-    await wait(500);
+    await processInstanceHandler.waitForProcessInstanceToReachSuspendedTask(correlationId, processModelReceiveEvents);
```
Comment (robpal94) [inline test/1_process_engine/interprocess_communication_test.js]:
```suggestion
    await processInstanceHandler
      .waitForProcessInstanceToReachSuspendedTask(correlationId, processModelReceiveEvents);
```

File: test/1_process_engine/interprocess_communication_test.js [hunk 2]
```diff
@@ -77,7 +81,7 @@ describe('Inter-process communication - ', () => {
     // As a result we must subscribe to the event that gets send when the test is done.
     testFixtureProvider.executeProcess(processModelReceiveEvents, 'SignalStartEvent_1', correlationId);
 
-    await wait(500);
+    await processInstanceHandler.waitForProcessInstanceToReachSuspendedTask(correlationId, processModelReceiveEvents);
```
Comment (robpal94) [inline test/1_process_engine/interprocess_communication_test.js]:
```suggestion
    await processInstanceHandler
      .waitForProcessInstanceToReachSuspendedTask(correlationId, processModelReceiveEvents);
```

File: test/1_process_engine/interprocess_communication_test.js [hunk 3]
```diff
@@ -103,7 +107,7 @@ describe('Inter-process communication - ', () => {
     const endEventToWaitFor = 'EndEvent_1';
     testFixtureProvider.executeProcess(processModelReceiveTask, 'StartEvent_1', correlationId);
 
-    await wait(500);
+    await processInstanceHandler.waitForProcessInstanceToReachSuspendedTask(correlationId, processModelReceiveTask);
```
Comment (robpal94) [inline test/1_process_engine/interprocess_communication_test.js]:
```suggestion
    await processInstanceHandler
      .waitForProcessInstanceToReachSuspendedTask(correlationId, processModelReceiveTask);
```

**Comment by ElRaptorus** [comment]
@robpal94 done

**Pull request merged**